src/test/java/org/passay/LengthComplexityRuleTest.java

```diff
@@ -0,0 +1,348 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import org.testng.AssertJUnit;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Unit test for {@link LengthComplexityRule}.
+ *
+ * @author  Middleware Services
+ */
+public class LengthComplexityRuleTest extends AbstractRuleTest
+{
+
+  /** Test password. */
+  private static final String INVALID_PASS_5 = "r%vE2";
+
+  /** Test password. */
+  private static final String VALID_PASS_9 = "r%scvEW2e";
+
+  /** Test password. */
+  private static final String VALID_PASS_12 = "rkscvEW2e93C";
+
+  /** Test password. */
+  private static final String VALID_PASS_18 = "rkscvEWbePwCOUovqt";
+
+  /** Test password. */
+  private static final String VALID_PASS_20 = "horse staple battery";
+
+  /** Test password. */
+  private static final String VALID_PASS_24 = "it was the best of times";
+
+  /** Test password. */
+  private static final String INVALID_PASS_78 =
+    "It was the best of times, it was the worst of times, it was the age of wisdom,";
+
+  /** For testing. */
+  private final LengthComplexityRule rule1 = new LengthComplexityRule();
+
+
+  /** Initialize rules for this test. */
+  @BeforeClass(groups = {"passtest"})
+  public void createRules()
+  {
+    rule1.setRules(
+      "[0,12)",
+      new LengthRule(8, 64),
```

Comment:
I was surprised by the presence of an explicit length rule, but on further thought it makes sense. I wonder if the range should be tied to the length rule in some way. You should probably cover all possible password lengths with your ranges; if you buy that then the start range should be `[8,12)` and end range should be `[20,64]`.